app/helpers/katello/katello_urls_helper.rb

```diff
@@ -1,15 +1,36 @@
 require 'uri'
+
 module Katello
   module KatelloUrlsHelper
+    def foreman_settings_url(schema = nil)
+      if schema
+        Settings[:foreman_url].sub(%r|^.*:|, "#{schema}:")
+      else
+        Setting[:foreman_url].sub(/^https/, 'http')
+      end
+    end
+
     def subscription_manager_configuration_url(host = nil, rpm = true)
       prefix = if host && host.content_source
-                 "http://#{@host.content_source.hostname}"
+                 "http://#{host.content_source.hostname}"
                else
-                 Setting[:foreman_url].sub(/\Ahttps/, 'http')
+                 foreman_settings_url
                end
       config = rpm ? SETTINGS[:katello][:consumer_cert_rpm] : SETTINGS[:katello][:consumer_cert_sh]
 
       "#{prefix}/pub/#{config}"
     end
+
+    def repository_url(content_path, content_type = 'repos', schema = 'http')
+      url = if @host.content_source
+              "#{schema}://#{@host.content_source.hostname}"
+            else
+              foreman_settings_url(schema)
+            end
+      path = ::Katello::Glue::Pulp::Repos.repo_path_from_content_path(
+        @host.lifecycle_environment, content_path)
+      return content_path if content_path =~ %r|^([\w\-\+]+)://|
```

Comment:
I think this should be the first line of the method, to not do some unnecessary logic wen ful url is provided.